agent/src/api/v1/storage.rs

```diff
@@ -136,6 +138,44 @@ async fn has_content_address(Path(address): Path<String>) -> impl IntoResponse {
     }
 }
 
+async fn patch_content_at_address(Path(address): Path<String>, body: Bytes) -> impl IntoResponse {
+    metrics::increment_counter!("storage:cas:patch");
+    let Some(storage) = STORAGE.get() else {
+        return (StatusCode::SERVICE_UNAVAILABLE, "storage uninitialized; programmer error".to_string()).into_response();
+    };
+
+    let Ok(integrity) = address.parse::<Integrity>() else {
+        let e = ServalError::BlobAddressInvalid(format!("{} is not a valid sub-resource integrity string", address));
+        return e.into_response()
+    };
+
+    match storage.data_by_integrity(integrity).await {
```

Comment:
Just noting for my own edification: I think the implication of this is that we can currently only patch things that will fit in memory, but that seems like a problem that won't bite us any time soon.